### ISSUE TITLE
Trouble adding all classes necessary to the classpath

### ISSUE THREAD

**Issue by DarthEmpty** [soot-oss/soot#606]
Hey, there

I'm rather new to Soot so I apologise if this question turns out to be a trivial one but I'm struggling to add both the path to "rt.jar" and the path to the directory containing the java files I wish to process to Soot's classpath.

I have java files that I want to process into jimple in a directory called "temp". Below is the command tried first:

```
java -jar soot-2.5.0.jar -allow-phantom-refs -cp temp -process-dir temp -d soot-output -src-prec java" -f J
```

Having read the wiki page "Introduction: Soot as a command line tool", I thought it best to add "temp" to Soot's classpath since that would grant Soot access to the java files to be processed. There were many phantom classes declared and then a NullPointerException was thrown.

I then tried the command below in an attempt to add "rt.jar" to the classpath:

```
java -jar soot-2.5.0.jar -allow-phantom-refs -cp "C:\Program Files\Java\jdk1.8.0_74\jre\lib\rt.jar" -process-dir temp -d soot-output -src-prec java" -f J
```

This allowed Soot to run all the way through and only a few classes were declared phantom. However, the list of phantom classes included the classes that I was trying to process meaning that the result was a directory of jimple files containing empty classes.

Any help would be much appreciated, Thank you very much for your time.

**Comment by StevenArzt** [comment]
Your second command line looks correct. In general, the Soot classpath should contain the classes you do not want to transform, but that are references by the classes you want to transform. They are only loaded on demand. In the process-dir, on the other hand, there are the classes you want to transform,

Note that your directory structure must be in Java format. If you have a class a.b.c, your code file must be located in a/b/c.java.

You seem to be using Soot 2.5 which is pretty outdated by now. As we don't have a newer release version (shame on us, we never really got to it), I would recommend to use our nightly builds available at https://ssebuild.cased.de/nightly/soot/.

**Comment by DarthEmpty** [comment]
Thank you for replying so quickly!

So, I tried again with the second command line and the most recent Soot nightly build today and I found that the classes that I wish to process are unfortunately, again, declared phantom by Soot, causing Soot's output to just be empty jimple classes. I'm certain that my files are stored in the correct places as you mentioned before (unless they're not allowed to be stored in "temp"?) and I'm not sure what I should be doing next.

Also, there seems to be an issue with the option "-process-dir" in this build. Using the same options as I did in the second command line, I received a NullPointerException that seems to occur when parsing classes. This happens without even providing warnings about phantom classes. Running Soot on one class, however, allows Soot to run to completion - it just results in the issue mentioned in the previous paragraph.

For reference, the command I've tried is below.
I've placed Soot in its own directory: "Soot".
I now use the option "-pp" seeing as it seems to work as intended.
"a" is one of the java classes I wish to transform.

```
java -jar soot\soot-trunk.jar -allow-phantom-refs -pp -d soot-output -src-prec java -f J temp\a
```

Again, thanks in advance.
